### PR TITLE
gitlab-release-cli: init at 0.18.0

### PR DESCRIPTION
**Pull request by kilimnik**
## Description of changes

This program is needed to create releases on gitlab. It is also necessary to be included when using nix as a base image in a gitlab ci which does a release.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by getchoo** [comment]
Result of `nixpkgs-review pr 323147` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gitlab-release-cli</li>
  </ul>
</details>

**Review by getchoo — APPROVED** [review]

File: pkgs/by-name/gi/gitlab-release-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+buildGoModule rec {
+  pname = "gitlab-release-cli";
+  version = "0.18.0";
+
+  src = fetchFromGitLab {
+    owner = "gitlab-org";
+    repo = "release-cli";
+    rev = "v${version}";
+    sha256 = "sha256-CCSice/uMf2OfFNEpwwhX6A0wrSsC1v9XWEhAAwQRso=";
```
Comment (Aleksanaa) [inline pkgs/by-name/gi/gitlab-release-cli/package.nix:14]:
```suggestion
    hash = "sha256-CCSice/uMf2OfFNEpwwhX6A0wrSsC1v9XWEhAAwQRso=";
```

File: pkgs/by-name/gi/gitlab-release-cli/package.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  };
+
+  vendorHash = "sha256-UwDMRsWbk8rEv2d5FssIzCLby68YZULoxd3/JGLsCQU=";
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/by-name/gi/gitlab-release-cli/package.nix:19]:
```suggestion
  meta = {
```

Better to avoid it.

Comment (kilimnik) [reply pkgs/by-name/gi/gitlab-release-cli/package.nix]: What's the reasoning, of why this should be avoided?

Comment (getchoo) [reply pkgs/by-name/gi/gitlab-release-cli/package.nix]: `lib` is a very broad scope, which when mixed with the [inherit issues](https://nix.dev/guides/best-practices#with-scopes) of `with ...;` can lead to a confusion at best and weird variable shadowing at worst. also see https://github.com/NixOS/nixpkgs/issues/208242

**Review by getchoo — APPROVED** [review]

File: pkgs/by-name/gi/gitlab-release-cli/package.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchFromGitLab {
+    owner = "gitlab-org";
+    repo = "release-cli";
+    rev = "v${version}";
+    hash = "sha256-CCSice/uMf2OfFNEpwwhX6A0wrSsC1v9XWEhAAwQRso=";
+  };
+
+  vendorHash = "sha256-UwDMRsWbk8rEv2d5FssIzCLby68YZULoxd3/JGLsCQU=";
+
```
Comment (Aleksanaa) [inline pkgs/by-name/gi/gitlab-release-cli/package.nix:18]:
```suggestion

doCheck = !stdenv.isDarwin;
```

Not sure why they are failing on darwin though...

File: pkgs/by-name/gi/gitlab-release-cli/package.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  meta = {
+    description = "GitLab toolset to create, retrieve and update releases";
```
Comment (afh) [inline pkgs/by-name/gi/gitlab-release-cli/package.nix:20]: Quoting the section [Meta attributes in `pkgs/README.md`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes):
> `meta.description` must:
> …
> ○ Not start with the package name.
>   ■ More generally, it should not refer to the package name.


```suggestion
    description = "Toolset to create, retrieve and update releases";
```

File: pkgs/by-name/gi/gitlab-release-cli/package.nix (hunk 3 above)
Comment (afh) [reply pkgs/by-name/gi/gitlab-release-cli/package.nix:18]: Looking at the build output I see

>  Error:          "failed to create release: failed to do request: Post \"https://127.0.0.1:56138/api/v4/projects/projectID/releases\": tls: failed to verify certificate: x509: “127.0.0.1” certificate is not standards compliant" does not contain "x509: certificate signed by unknown authority"

Are the checks dependent on external services that are not running in a NixOS build?
Or is the `$NIX_SSL_CERT_FILE` possibly not found/used?

**Review by afh — COMMENTED** [review]
Please find below my response to your comment, @Aleksanaa.

And a suggestion for improving `meta.description`, @kilimnik 🙂

File: pkgs/by-name/gi/gitlab-release-cli/package.nix (hunk 4 above)
Comment (Aleksanaa) [reply pkgs/by-name/gi/gitlab-release-cli/package.nix:20]: This change doesn't look appropriate. We should state that it applies to GitLab.

Comment (afh) [reply pkgs/by-name/gi/gitlab-release-cli/package.nix:20]: How is this, @Aleksanaa? 🙂 

```suggestion
    description = "Toolset to create, retrieve and update releases on GitLab";
```

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 323147` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>gitlab-release-cli</li>
  </ul>
</details>

File: pkgs/by-name/gi/gitlab-release-cli/package.nix (hunk 4 above)
Comment (Aleksanaa) [reply pkgs/by-name/gi/gitlab-release-cli/package.nix:20]: I'm not sure what the difference is between this change and the previous one. It even seems a bit more verbose

File: pkgs/by-name/gi/gitlab-release-cli/package.nix (hunk 3 above)
Comment (afh) [reply pkgs/by-name/gi/gitlab-release-cli/package.nix:18]: How about skipping the failing test individually until the underlying cause of the test failure on darwin is found? Note that if this suggestion is accepted `stdenv` needs to be added to the inputs (see suggestion above).
```suggestion
  checkFlags = lib.optionals stdenv.isDarwin [
    # Skip failing test
    "-skip TestHTTPSCustomCA"
  ];
```

File: pkgs/by-name/gi/gitlab-release-cli/package.nix [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{ lib
+, fetchFromGitLab
```
Comment (afh) [inline pkgs/by-name/gi/gitlab-release-cli/package.nix:2]: Needed to skip the failing test (see suggestion below)
```suggestion
, stdenv
, fetchFromGitLab
```

**Review by afh — COMMENTED** [review]
I'd like to suggest the changes below (see comments), and if you wanted to go above and beyond with this PR, @kilimnik  please consider formatting it using:
<code>nix run <a href='https://github.com/NixOS/nixfmt?tab=readme-ov-file#nixfmt'>github:NixOS/nixfmt</a> -- pkgs/by-name/gi/gitlab-release-cli/package.nix</code>

File: pkgs/by-name/gi/gitlab-release-cli/package.nix (hunk 5 above)
Comment (kilimnik) [reply pkgs/by-name/gi/gitlab-release-cli/package.nix]: Thanks for testing on darwin

File: pkgs/by-name/gi/gitlab-release-cli/package.nix (hunk 4 above)
Comment (kilimnik) [reply pkgs/by-name/gi/gitlab-release-cli/package.nix]: Thank you for the discussion,  I applied the last description.

**Comment by afh** [comment]
Result of `nixpkgs-review pr 323147` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gitlab-release-cli</li>
  </ul>
</details>

**Review by afh — APPROVED** [review]
Thanks for accepting the suggestions and making the changes, @kilimnik, much appreciated.

From my end this looks good to be merged 👍

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**